src/components/form/Form.js

```diff
@@ -98,12 +98,14 @@ export class FormComponent extends FormioForm {
    * Submit the form before the whole form is triggered.
    */
   beforeSubmit() {
-    // Before we submit, we need to filter out the references.
-    this.data[this.component.key] = this.component.reference ? {_id: this._submission._id} : this._submission;
-
     // Ensure we submit the form.
     if (this.component.submit && !this.submitted) {
-      return this.submit(true);
+      let submission = this.submit(true);
+
+      // Before we submit, we need to filter out the references.
+      this.data[this.component.key] = this.component.reference ? {_id: submission._id, form: submission.form} : submission;
+
+      return submission;
```

Comment:
Oh... this may be causing it.

You are returning the submission here, but this should return a promise. This is getting propagated up to the parent form which is why this seems to work, but I am sure this is a bug.

Reply:
I'm taking a second look at this to see what is going on.